src/api/checkin.js

```diff
@@ -0,0 +1,17 @@
+import express from 'express';
+import { addAction } from '../lib/database';
+
+let app = express();
+
+app.post('/', (req, res) => {
+    addAction('CHECKIN', {
+        studentId: req.body.studentId,
+        itemAddress: req.body.itemAddress
+    }).then(() => {
+        res.successJson();
+    }).catch(e => {
```

Comment:
This will catch (unlikely) errors in writing the action to the database, but it will not catch logical errors from the checkin store (e.g. item is already checked in, or student does not exist). To do this, try changing the `then() => {` to `then(actionId => {` and using `getCheckinByActionId` and `getCheckinErrorByActionId` from the CheckinStore to check for success/failure.

I can help with this if it is unclear. I could also start the unhappy paths task for checkout, as it will need to work in the same way.